src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -180,58 +182,77 @@ public int Seek(Vector<byte> byte0Vector)
 
             var block = _block;
             var index = _index;
-            var array = block.Array;
+            var following = block.End - index;
+            byte[] array;
+            var byte0 = byte0Vector[0];
+
             while (true)
             {
-                while (block.End == index)
+                while (following == 0)
                 {
-                    if (block.Next == null)
+                    var newBlock = block.Next;
+                    if (newBlock == null)
                     {
                         _block = block;
                         _index = index;
                         return -1;
                     }
-                    block = block.Next;
-                    index = block.Start;
-                    array = block.Array;
+                    index = newBlock.Start;
+                    following = newBlock.End - index;
+                    block = newBlock;
                 }
-                while (block.End != index)
+                array = block.Array;
+                while (following > 0)
                 {
-                    var following = block.End - index;
-                    if (following >= Vector<byte>.Count)
+#if !DEBUG 
+// Need unit tests to test Vector path
+                    // Check will be Jitted away https://github.com/dotnet/coreclr/issues/1079
+                    if (Vector.IsHardwareAccelerated)
                     {
-                        var data = new Vector<byte>(array, index);
-                        var byte0Equals = Vector.Equals(data, byte0Vector);
-
-                        if (byte0Equals.Equals(Vector<byte>.Zero))
+#endif
+                        if (following >= _vectorSpan)
```

Comment:
K, new change seems to resolve it `VectorSpan` is a const `10h`
```
   211:                         if (following >= Constants.VectorSpan)
00007FFE8A62C137  cmp         ebx,10h  
00007FFE8A62C13A  jl          00007FFE8A62C1D4  
00007FFE8A62C140  mov         ecx,dword ptr [r15+8]  
00007FFE8A62C144  cmp         edi,ecx  
00007FFE8A62C146  jae         00007FFE8A62C23D  
00007FFE8A62C14C  lea         r8d,[rdi+0Fh]  
00007FFE8A62C150  cmp         r8d,ecx  
00007FFE8A62C153  jae         00007FFE8A62C23D  
00007FFE8A62C159  movups      xmm0,xmmword ptr [r15+rdi+10h]  
```
Also the `BitConverter.IsLittleEndian` test and static initialization is removed
```
283:             knownString = null;
00007FFE8A62D6E2  push        rbx  
00007FFE8A62D6E3  sub         rsp,20h  
00007FFE8A62D6E7  mov         rsi,rcx  
00007FFE8A62D6EA  mov         rdi,rdx  
00007FFE8A62D6ED  mov         rbx,r8  
00007FFE8A62D6F0  mov         rcx,7FFE8A720B28h  
00007FFE8A62D6FA  mov         edx,22h  
00007FFE8A62D6FF  call        00007FFEEA30BC50  
00007FFE8A62D704  xor         ecx,ecx  
00007FFE8A62D706  mov         qword ptr [rbx],rcx  
   289:             }
   290: 
   291:             var inputLength = begin.GetLength(end);
00007FFE8A62D709  mov         rcx,rsi  
00007FFE8A62D70C  mov         rdx,rdi  
00007FFE8A62D70F  call        00007FFE8A5F95D0  
00007FFE8A62D714  mov         edi,eax  
```